include/deal.II/matrix_free/fe_point_evaluation.h

```diff
@@ -853,14 +904,45 @@ FEPointEvaluation<n_components, dim, spacedim, Number>::reinit(
       fe_values->reinit(cell);
     }
 
-  this->unit_points =
-    std::vector<Point<dim>>(unit_points.begin(), unit_points.end());
+  this->unit_points = unit_points;
+
+  const_cast<unsigned int &>(n_q_points) = unit_points.size();
```

Comment:
I see why you do this. I do the same in `FEFaceValues` to support mixed meshes. The reason for me to do that there was because `n_q_points` was already a const and public member variable. I was not happy with the solution there, since it is very error prone on user side if one should decide to store `n_q_points` in a variable. I would much more prefer if we could either introduce a getter function or, if we just need this to iterate over quadrature points, a function like `quadrature_point_indices()` in `FEEvaluation` (https://www.dealii.org/developer/doxygen/deal.II/classFEEvaluationData.html#a0eca800437c07c9f23ec6815e58f1432).